File: 1260-shift-2d-grid/1260-shift-2d-grid.py
class Solution(object):
    def shiftGrid(self, grid, k):
    
        m = len(grid)
        n = len(grid[0])

        arr = []
        for row in grid:
            arr.extend(row)

        total = m * n
        k %= total
        arr = arr[-k:] + arr[:-k]

        ans = []
        for i in range(0, total, n):
            ans.append(arr[i:i+n])

        return ans

        
        """
        :type grid: List[List[int]]
        :type k: int
        :rtype: List[List[int]]
        """
        

File: 1260-shift-2d-grid/1260-shift-2d-grid.js
/**
 * @param {number[][]} grid
 * @param {number} k
 * @return {number[][]}
 */
var shiftGrid = function(grid, k) {
    const m = grid.length;
    const n = grid[0].length;
    const total = m * n;

    let arr = [];
    for (let row of grid) {
        arr.push(...row);
    }

    k %= total;
    arr = arr.slice(total - k).concat(arr.slice(0, total - k));

    let ans = [];
    for (let i = 0; i < total; i += n) {
        ans.push(arr.slice(i, i + n));
    }

    return ans;
    
};